src/Nox.Types/Types/Volume/Volume.cs

```diff
@@ -8,46 +9,130 @@ namespace Nox.Types;
 /// </summary>
 public class Volume : Measurement<Volume, VolumeUnit>
 {
+    public const ushort QuantityValueDecimalPrecision = 6;
+
+    private VolumeTypeOptions _volumeTypeOptions = new();
+    private readonly VolumeTypeUnit _unit;
+
+    public new VolumeTypeUnit Unit
+    {
+        get => _unit;
+        private init
+        {
+            _unit = value;
+            base.Unit = Enumeration.ParseFromName<VolumeUnit>(_unit.ToString());
+        }
+    }
+
     /// <summary>
-    /// Creates a new instance of <see cref="Volume"/> object in cubic feet.
+    /// Creates a new instance of <see cref="Volume"/> object in cubic meters.
     /// </summary>
-    /// <param name="value">The value to create the <see cref="Volume"/> with</param>
+    /// <param name="value">The origin value to create the <see cref="Volume"/> with</param>
     /// <returns></returns>
     /// <exception cref="TypeValidationException"></exception>
-    public static Volume FromCubicFeet(QuantityValue value)
-        => From(value, VolumeUnit.CubicFoot);
+    public new static Volume From(QuantityValue value)
+        => From(value, new VolumeTypeOptions());
 
     /// <summary>
-    /// Creates a new instance of <see cref="Volume"/> object in cubic meters.
+    /// Creates a new instance of <see cref="Volume"/> object in specified unit.
     /// </summary>
-    /// <param name="value">The origin value to create the <see cref="Volume"/> with</param>
+    /// <param name="value">Value to create a <see cref="Volume"/>.</param>
+    /// <param name="unit">Unit to create a <see cref="Volume"/>.</param>
     /// <returns></returns>
     /// <exception cref="TypeValidationException"></exception>
-    public static Volume FromCubicMeters(QuantityValue value)
-        => From(value, VolumeUnit.CubicMeter);
+    public new static Volume From(QuantityValue value, VolumeUnit unit)
```

Comment:
We don't want to expose VolumeUnit in the construction method, so please remove this one and mark VolumeUnit and VolumeConversion as internal.

Reply:
Method removed.